js/main.js

```diff
@@ -50,6 +51,47 @@ $entryForm.addEventListener('submit', function () {
   data.editing = null;
 });
 
+// here start the delete process
+
+var $deleteTag = document.querySelector('.deleteTag');
+var $modalHolder = document.querySelector('.modalHolder');
+var $cancel = document.querySelector('.cancel');
+var $confirm = document.querySelector('.confirm');
+var modal = false;
+
+function callModal(event) {
+  modal = !modal;
+  if (modal === true) {
+    $modalHolder.className = ' modalHolder';
+  } else {
+    $modalHolder.className = ' modalHolder hidden';
+  }
+}
+
+$confirm.addEventListener('click', function () {
+
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries[i].entryId === data.editing.entryId) {
+      data.entries.splice(i, 1);
+
+      var $allEntry = document.querySelectorAll('.allEntries');
+
+      for (var j = 0; j < $allEntry.length; j++) {
+        if ($allEntry[j].getAttribute('data-entry-id') === String(data.editing.entryId)) {
+          $allEntry[j].remove();
+        }
+      }
+    }
+  }
+
+  viewChange(event.target.getAttribute('data-view'));
+  callModal(event);
+  $deleteTag.className = 'deleteTag noShow';
+});
```

Comment:
Instead of making this event handler an anonymous function, I would make it a named function with a name such as `deleteEntry`